Another attempt at M1 compilation

kustosz opened the pull request:
### Pull Request Description

<!--
- Please describe the nature of your PR here, as well as the motivation for it.
- If it fixes an open issue, please mention that issue number here.
-->

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [ ] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

kustosz committed:
set outdir

kustosz committed:
[no-changelog] no changelog

JaroslavTulach commented on build.sbt:
```diff
@@ -680,9 +680,20 @@ val generateRustParserLib =
+        "x86_64-apple-darwin",
+        "-Z",
+        "unstable-options",
+        "--out-dir",
```
That's good idea to share the output dir!

JaroslavTulach commented on project/DistributionPackage.scala:
```diff
@@ -136,11 +136,7 @@ object DistributionPackage {
     )
     val os = System.getProperty("os.name")
     val isMac = os.startsWith("Mac")
-    val dir = if (isMac) {
```
Thanks for this simplification. I hope it works!

JaroslavTulach reviewed (approved):
Thanks a lot Marcin, I have just confirmed myself that your changes work well on my Intel based Mac. Please integrate as soon as possible. Otherwise integration of #3611 breaks all developers compiling on Mac M1.

hubertp commented on build.sbt:
```diff
@@ -680,9 +680,20 @@ val generateRustParserLib =
+        "-Z",
+        "unstable-options",
+        "--out-dir",
+        "target/rust/debug"
```
This shouldn't be hard-coded. Instead one should use the value of the output target setting.

hubertp commented on lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java:
```diff
@@ -27,8 +27,7 @@ public final class Parser implements AutoCloseable {
       File root = new File(".").getAbsoluteFile();
-      if (!searchFromDirToTop(e, root, "target", "rust", "x86_64-apple-darwin", "debug", name)
-          && !searchFromDirToTop(e, root, "target", "rust", "debug", name)) {
+      if (!searchFromDirToTop(e, root, "target", "rust", "debug", name)) {
```
Similarly as in the other case, those shouldn't be hardcoded.

hubertp replied:
OK, this is in static initializer so it will be hard to configure. Note that the `debug` bit is probably gonna be problematic for non-dev releases

hubertp commented on project/DistributionPackage.scala:
```diff
@@ -136,11 +136,7 @@ object DistributionPackage {
-    } else {
-      "target/rust/debug/"
-    }
+    val dir = "target/rust/debug/"
```
This  value should be provided from the outside, hard-coding this is probably a bad idea

hubertp reviewed (changes requested):
Try to use sbt's setting rather than hardcoding the target path in various places

hubertp committed:
Use SettingsKey instead of hardcoding target dir

Left the non-dev version unknown (???) since I don't know how is that
supposed to work under such circumstances.

hubertp reviewed (approved):
@kustosz pushed my change with SettingsKey to illustrate what I had in my mind when talking about hardcoded values. Hope you don't mind

kustosz committed:
refactors

kustosz committed:
Merge branch 'develop' into wip/mk/mac-build
